todo-app/src/TextField.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+
+
+export default class TextField extends React.Component {
```

Comment:
This needs to be two components, not one. It's fine to render either a textarea or a single-line input from a component with configurable props, but rendering a date input from a component called `TextField` does not make any sense.